### PR TITLE
[MetaSchedule] Fix thread bindings of MultiLevelTilingTensorCore

### PR DESCRIPTION
**Pull request by vinx13**
Fixed https://github.com/apache/tvm/issues/13204.

cc @junrushao

### CODE REVIEW

**Review by michalpiszczek — APPROVED** [review]

**Review by shingjan — APPROVED** [review]
LGTM. Thanks!

**Review by junrushao — APPROVED** [review]
LGTM!

**Comment by junrushao** [comment]
BTW, i was curious why it's not caught by the unittests of search space generation?

**Comment by vinx13** [comment]
the unit test doesn't use default config here

**Comment by junrushao** [comment]
we might want to add a unittest to cover the rule, but i understand it's kind of urgent. please set a reminder to add a regression test :-)

**Comment by masahi** [comment]
@tvm-bot rerun

File: src/meta_schedule/schedule_rule/multi_level_tiling_tensor_core.cc [hunk 1]
```diff
@@ -556,6 +556,11 @@ ScheduleRule ScheduleRule::MultiLevelTilingTensorCore(
     Optional<Integer> max_innermost_factor, Optional<Array<Integer>> vector_load_lens,
     Optional<Map<String, ObjectRef>> reuse_read, Optional<Map<String, ObjectRef>> reuse_write,
     bool use_software_pipeline) {
+  if (tile_binds.defined()) {
+    for (const String& tile_bind : tile_binds.value()) {
+      CHECK_NE(tile_bind, "threadIdx.x") << "Cannot bind to threadIdx.x when using tensor core.";
+    }
+  }
```
Comment (csullivan) [inline src/meta_schedule/schedule_rule/multi_level_tiling_tensor_core.cc:563]: Does this check (along with existing tests) now cover the regression this fixes? If not adding a test to protect against the regression in the future will surely save you and others time!

Comment (csullivan) [reply src/meta_schedule/schedule_rule/multi_level_tiling_tensor_core.cc:563]: Oops looks like @junrushao already asked my question, I trust the your judgements :)

Comment (vinx13) [reply src/meta_schedule/schedule_rule/multi_level_tiling_tensor_core.cc:563]: Yes this check covered the broken case that incorrectly using `threadIdx.x`. As a follow up, I'm thinking also adding some e2e search space generation test for the default config for each target.

**Pull request merged**